main.go

```diff
@@ -169,26 +175,32 @@ func main() {
 	}
 }
 
-func prepareEventsToTrace(eventsToTrace []string, excludeEvents []string) ([]int32, error) {
+func prepareEventsToTrace(eventsToTrace []string, tagsToTrace []string, excludeEvents []string) ([]int32, error) {
 	var res []int32
 	eventsNameToID := make(map[string]int32, len(tracee.EventsIDToEvent))
 	for _, event := range tracee.EventsIDToEvent {
 		eventsNameToID[event.Name] = event.ID
 	}
-	if eventsToTrace == nil {
+	tagsToEvents := make(map[string][]int32)
+	for id, event := range tracee.EventsIDToEvent {
+		for _, tag := range event.Tags {
+			tagsToEvents[tag] = append(tagsToEvents[tag], id)
+		}
+	}
+	if eventsToTrace == nil && tagsToTrace == nil {
+		isExcluded := make(map[int32]bool)
 		for _, name := range excludeEvents {
 			id, ok := eventsNameToID[name]
 			if !ok {
 				return nil, fmt.Errorf("invalid event to exclude: %s", name)
 			}
-			event := tracee.EventsIDToEvent[id]
-			event.EnabledByDefault = false
-			tracee.EventsIDToEvent[id] = event
+			isExcluded[id] = true
 		}
 		res = make([]int32, 0, len(tracee.EventsIDToEvent))
-		for _, event := range tracee.EventsIDToEvent {
-			if event.EnabledByDefault {
-				res = append(res, event.ID)
+		defaultEvents, _ := tagsToEvents["default"]
+		for _, id := range defaultEvents {
+			if !isExcluded[id] {
+				res = append(res, id)
```

Comment:
I think that instead of special casing "default", you can just set `tagsToTrace = "default` if `eventsToTrace == nil` and `tagsToTrace == nil`, and the process the tags as as usual

Reply:
I thought about this too, the problem his is that you loose the mutual exclusion with "excluded-events", which is indeed a special case of the default behavior

Reply:
why is this a special case of the default event-set? for example `--events-set network --exclude-event socket seems` like a legit use case no?

Reply:
Yes, see my other comments below, as these two are related